azure-quantum/azure/quantum/target/params.py

```diff
@@ -0,0 +1,243 @@
+        elif type(value) == list:
+            raise TypeError(f"Nested lists are not supported")
+        else:
+            type_name = type(value).__name__
+            raise TypeError(f"Unsupported type {type_name} for {value}")
+
+
+@dataclass
+class AutoValidatingParams:
+    """
+    A helper class for target parameters.
+
+    It has a function as_dict that automatically extracts a dictionary from
+    the class' fields.  They are added to the result dictionary if their value
+    is not None, the key is automatically transformed from Python snake case
+    to camel case, and if validate is True and if the field has a validation
+    function, the field is validated beforehand.
+    """
+    def as_dict(self, validate=True):
+        result = {}
+
+        for name, field in self.__dataclass_fields__.items():
+            field_value = self.__getattribute__(name)
+            if field_value is not None:
+                # validate field?
+                if validate and "validate" in field.metadata:
+                    field.metadata["validate"].__func__(name, field_value)
+
+                # translate field name to camel case
+                s = sub(r"(_|-)+", " ", name).title().replace(" ", "")
+                attribute = ''.join([s[0].lower(), s[1:]])
+                result[attribute] = field_value
+
+        if validate:
+            self.post_validation(result)
+
+        return result
+
+    def post_validation(self, result):
+        """
+        A function that is called after all individual fields have been
+        validated, but before the result is returned.
+
+        Here result is the current dictionary.
+        """
+        pass
+
+def validating_field(validation_func, default=None):
+    """
+    A helper method to declare field for an AutoValidatingParams data class.
+    """
+    return field(default=default, metadata={"validate": validation_func})
+
+
+class InputParamsItem:
+    """
+    Base class for input parameters.
+
+    This class serves both as the base class for InputParams as well as for the
+    items in the InputParams.
+    """
+    def __init__(self):
+        # all input param items may have an entry point name and a list of
+        # arguments
+        self.entry_point: Optional[str] = None
+        self.arguments = EntryPointArguments()
+
+    def as_dict(self, validate=True) -> Dict[str, Any]:
+        """
+        Returns input params as a dictionary.
+        """
+        result = {}
+
+        if self.entry_point is not None:
+            result['entryPoint'] = self.entry_point
+
+        if len(self.arguments.entries) > 0:
+            result['arguments'] = self.arguments.entries
+
+        return result
+
+
+class InputParams(InputParamsItem):
+    """
+    Class to define input parameters.
+
+    This class allows to define inout parameters for non-batching and batching
+    jobs.  The instance represents a batching job, if and only if num_items is
```

Comment:
I don't think we expose the notion of a "batching job" anywhere externally, so I think it is best to avoid any references in documentation

Reply:
We do expose it here in the documentation: https://learn.microsoft.com/en-us/azure/quantum/how-to-work-with-re#batching-with-q